conans/test/integration/lockfile/test_lock_requires.py

```diff
@@ -279,3 +280,83 @@ def requirements(self):
     assert "dep/0.2#" in client.out
     assert "dep/0.1" not in client.out
     assert "dep/0.3" not in client.out
+
+
+def test_locking_require_override():
+    """
+    conanfile.txt locking it dependencies and using a ``--require-override``
+    that forces an override over a transitive dependency, without version ranges.
+    Then, when the lockfile is applied, it will fail, because the conanfile not overriden
+    dependency doesn't match the locked one
+    """
+    client = TestClient()
+    client.save({"pkg/conanfile.py": GenConanfile(),
+                 "consumer/conanfile.txt": f"[requires]\npkg/0.1"})
+    client.run("create pkg --name=pkg --version=0.1")
+    client.run("create pkg --name=pkg --version=0.2")
+    client.run("lock create consumer/conanfile.txt  --lockfile-out=conan.lock "
+               "--require-override=pkg/0.2")
+    assert "pkg/0.2" in client.out
+    assert "pkg/0.1" not in client.out
+
+    # FIXME: This fails, as the lockfile contains pkg/0.2, and conanfile pkg/0.1 cannot be
+    #  found in lockfile
+    client.run("install consumer/conanfile.txt --lockfile=conan.lock --lockfile-strict",
+               assert_error=True)
+    assert "ERROR: Requirement 'pkg/0.1' not in lockfile" in client.out
```

Comment:
This is not an easy behavior, I would love to simplify the requires overrides. Let's discuss next week what can we do.

Reply:
Removed the test, leaving just:

- the normalization of ``--lockfile-strict``
- a small refactor to add the cli args in a common place